core/src/main/java/io/grpc/transport/AbstractStream.java

```diff
@@ -150,6 +186,11 @@ public void dispose() {
   }
 
   /**
+   * Gets the listener to this stream.
+   */
+  protected abstract LT listener();
```

Comment:
Just make this return StreamListener? It doesn't seem we need it to be generic at all, or did I miss something?